go-controller/pkg/ovn/master.go

```diff
@@ -811,6 +821,15 @@ func (oc *DefaultNetworkController) addUpdateNodeEvent(node *kapi.Node, nSyncs *
 	return err
 }
 
+func (oc *DefaultNetworkController) addUpdateRemoteNodeEvent(node *kapi.Node) error {
+	_, present := oc.localZoneNodes.Load(node.Name)
+
+	if present {
```

Comment:
when will we have a situation where `addUpdateRemoteNodeEvent` is called with node present in local zone as per cache? we seem to always check the `isLocalZone` thing before calling this function. I am a bit confused on when and where the actual cache gets updated when a node changes zone.

Reply:
The cache gets updated in the functions 
  - addUpdateLocalNodeEvent  -> the node  name is added to the localZoneNodes cache
  - addUpdateRemoteNodeEvent -> the node name is deleted from the cache if its present
  - deleteNodeEvent -> the node name is deleted from the cache if its present.

The caller (default_network_controller.go) calls isLocalZone() on both the old node object and new node object to determine
   - If the node is moved from a remote zone to the local zone
         -> In this case,  isLocalZone(oldNode) will be "false" and isLocalzone(newNode) will be "true"

   - If the node is moved from the local zone to a remote zone
        -> In this case,  isLocalZone(oldNode)  will be "true" and isLocalzone(newNode) will be "false"